modules/product/src/main/java/com/opengamma/strata/product/etd/EtdStyleType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.etd;
+
+import org.joda.convert.FromString;
+import org.joda.convert.ToString;
+
+import com.google.common.base.CaseFormat;
+import com.opengamma.strata.collect.ArgChecker;
+
+/**
+ * The type of an Exchange Traded Derivative (ETD) product, either a future or an option.
+ */
+public enum EtdStyleType {
```

Comment:
But it's a name that doesn't have any real-world meaning. `EtdStyleType` means nothing to someone with domain knowledge. `EtdContractType` is a pretty meaningful description of what it is. 

I thought of `EtdContractType` after looking at the Javadoc for each of the fields and seeing they call contain "contract".